frontend/app/src/types/user.ts

```diff
@@ -181,11 +181,28 @@ export const Tag = z.object({
   name: z.string(),
   description: z.string(),
   backgroundColor: z.string(),
-  foregroundColor: z.string()
+  foregroundColor: z.string(),
+  readOnly: z.boolean().default(false).optional(),
+  icon: z.string().default('').optional()
 });
 
 export type Tag = z.infer<typeof Tag>;
 
+export enum ReadOnlyTag {
+  LOOPRING = 'Loopring'
+}
+
+export const READ_ONLY_TAGS: { [key: string]: Tag } = {
```

Comment:
any chance we would use [mapped types](https://www.typescriptlang.org/docs/handbook/2/mapped-types.html) to limit the possible key values?

Reply:
Updated @kelsos